src/app/(user)/items/page.tsx

```diff
@@ -0,0 +1,25 @@
+"use client";
+
+import ProductItem from "@/components/product-item";
+import { productsInStock } from "@/domains/product/productsInStock";
+import { productsToPreOrder } from "@/domains/product/productsToPreOrder";
+import { useSearchParams } from "next/navigation";
+
+export default function Items() {
+  const allProducts = [...productsInStock, ...productsToPreOrder];
+  const searchParams = useSearchParams();
+
+  return (
+    <div className="section-container">
+      <div className="flex flex-col flex-wrap gap-12">
+        {allProducts
+          .filter((p) => p.category === searchParams.get("category"))
```

Comment:
If you want to use the same query parameter everytime, it's better to extract it in the outer block:
```tsx
const searchParams = useSearchParams();
const activeCategory = searchParams.get("category")

...
allProducts.filter((p) => p.category === activeCategory)
```